### ISSUE TITLE
blank page login, qs bug from mapped custom domain Uncaught SyntaxError: Unexpected string

### ISSUE THREAD

**Issue by jjhesk** [keystonejs/keystone#3979]
resurrected issue from #3642

### Steps to reproduce the behavior
just login

### Expected behavior
can see the login screen
### Actual behavior

looks like i have the same issue with this 4.0.0-beta5 deploy @jstockwin @joeinnes @conrad-vanl on heroku. the issue doesnt appear on local deploy from testing server Ubuntu-16.4.10 tho. currently is not working.. http://bms.heskeyo.com/admiral/signin

Uncaught SyntaxError: Unexpected string
..

s)"undefined"!=typeof r&&n(r)}}var t=this,n=this.store.dispatch
Uncaught Error: Cannot find module 'qs'

**Comment by bassjacob** [comment]
I'm not sure what would be causing qs not to be requireable, it's a direct dependency of keystone. Can you run `npm cache clear; rm -rf node_modules; npm i` and see if that fixes the issue?

**Comment by jjhesk** [comment]
@bassjacob according to the local test project and the project runs fine without any errors. the only issue happens only on heroku deployment. note that the deployment will clear all cache for each compile deploys. I will check with ur notes to do so and see if there are any chance to get things clear out..

**Comment by jjhesk** [comment]
@bassjacob 
here is what i get when i ran `npm ls qs`
```
galleria-loopback-a73@2.3.5 /app
+-- body-parser@1.15.2
| `-- qs@6.2.0 
+-- loopback@3.1.0
| +-- loopback-datasource-juggler@3.1.0
| | `-- qs@3.1.0 
| `-- strong-remoting@3.1.1 (git+https://github.com/strongloop/strong-remoting.git#35fd6690b73838fde6925c0f99186afaf5f4102e)
|   `-- qs@6.3.0 
+-- loopback-component-storage@3.0.0-alpha.1 (git+https://github.com/strongloop/loopback-component-storage.git#c73dbc43b6f20113708353f5b678d63176f8603d)
| `-- pkgcloud@1.3.0
|   +-- qs@1.2.2 
|   `-- request@2.40.0
|     `-- qs@1.0.2 
+-- loopback-datasource-juggler@2.53.0
| `-- qs@3.1.0 
+-- request@2.76.0
| `-- qs@6.3.0 
`-- stripe@4.14.0 (git+https://github.com/stripe/stripe-node.git#d2660aff03851f2c259b9419323c1552ff9c484d)
  `-- qs@2.4.2 


```

**Comment by jjhesk** [comment]
@bassjacob this issues finally is discovered by using custom domain with https.
The original route was from https://gallerygoadmin.herokuapp.com/ which is the original but it is failed at the mapped domain. any hints?

**Comment by bassjacob** [comment]
honestly, no idea. all I can think of is if there is a corrupted version of the packages.js file, but running `npm cache clear; rm -rf node_modules; npm i` should tell us if that's the case.

**Comment by jjhesk** [comment]
@bassjacob i have totally done that and I have removed the one with the older version ` qs@2.3.3 ` and it is not working. I suspected that if there has anything to do with the cert of the security connection made by keystone.

**Comment by ash-rain** [comment]
Any new info?
I'm having the same issue, but I'm on dedicated server. When running local it's ok, but for some reason since a few hours ago I get blank page and this same error in the browser console when trying to access Keystone.

**Comment by peterpme** [comment]
:wave: I'm running into this issue too!

**Comment by jjhesk** [comment]
@bassjacob @JedWatson @jossmac @webteckie @mxstbr @bladey anyone can take a look of this issue? this seems to be an urgent issue. many thanks.

**Comment by thejameskyle** [comment]
Can people provide more information about the versions of keystone and npm/yarn they are using to reproduce this?

**Comment by jjhesk** [comment]
@thejameskyle ok, here is the info of the versions
```
{
  "name": "galleriokeystone",
  "version": "4.1.2",
  "private": true,
  "repository": {
    "type": "git",
    "url": ""
  },
  "engines": {
    "node": "6.9.2"
  },
  "dependencies": {
    "@risingstack/graffiti-mongoose": "^6.0.0",
    "async": "^2.0.0-rc.3",
    "crypto-js": "3.1.6",
    "csv": "^0.4.6",
    "dotenv": "2.0.0",
    "geoip-lite": "1.1.8",
    "graphql": "^0.9.1",
    "jwt-simple": "0.5.0",
    "keystone": "^4.0.0-beta.5",
    "keystone-live": "3.7.2",
    "lodash": "^4.16.4",
    "moment": "^2.12.0",
    "mongoose": "^4.8.1",
    "mongoose-deep-populate": "^3.0.0",
    "pug": "^2.0.0-beta9",
    "request-ip": "1.2.2",
    "restful-keystone-onode": "^0.2.9"
  },
  "scripts": {
    "start": "node keystone.js",
    "web": "node web"
  }
}

```

**Comment by anosikeosifo** [comment]
I'm still trying to find my way in this tunnel! Please, any light yet?

**Comment by anosikeosifo** [comment]
I've been able to get around it, here are my observations & solution:

There seems to be a mismatch between the asset requests path(src value) and the actual paths of those assets on disc, namely: signin.js, fields.js, admin.js.
It turns out that neither of these filenames exists.

The public folder in keystone/admin/public has only one bundled file(package.js), which loads normally, however when requests are made to the aforementioned files, we get a 404.
After a long tinkering, I noticed a similar file structure in keystone/admin/bundles.

So my solution was to write nginx alias commands to route asset requests as follows:
`[domain name]/keystone/js/signin.js > [app_location_on_disk]/keystone/admin/bundles/js/[some_number]-Signin/index.js`

`[domain name]/keystone/js/admin.js > [app_location_on_disk]/keystone/admin/bundles/js/[some_number]-App/index.js`

`[domain name]/keystone/js/fields.js > [app_location_on_disk]/keystone/admin/bundles/js/[some_number]-FieldsTypes.js`

I think these assets could be renamed to match the actual requests for them, or better still, be moved to the public directory.

**Comment by alonoz** [comment]
Same issue beta 4.0beta5, trying to debug:

signin.js:1 Uncaught Error: Cannot find module 'qs'

**Comment by anosikeosifo** [comment]
Yeah, that's same version i'm using.

**Comment by t100n** [comment]
I just stumbled upon this issue, and I was able to figure out what was causing my problem, so maybe it can help others with the same issue.

In my scenario the issue was due to a minification applied to all JS files by my CDN, removing a break line that exists after the e.apply():
{function e(e){return function(){var r=e.apply(t,arguments)"undefined"!=typeof r&&n(r)}}var t=this,n=this.store.dispatch

it should be:
{function e(e){return function(){var r=e.apply(t,arguments)
"undefined"!=typeof r&&n(r)}}var t=this,n=this.store.dispatch

**Comment by anosikeosifo** [comment]
@t100n cuold you help specifying what cdn, as well as the line number.

**Comment by t100n** [comment]
@anosikeosifo the CDN i'm using is Cloudflare.
The line in question is the 159 (in the original file), when the file is minified it triggers the "Uncaught SyntaxError: Unexpected string", and subsequent "Uncaught Error: Cannot find module 'qs'".

**Comment by Noviny** [comment]
Fixed by #4432. Please reopen fi you are still experiencing problems.

**Comment by Zedonboy** [comment]
Any solution yet?
